src/common/drpc.c

```diff
@@ -248,39 +235,35 @@ unixcomm_connect(char *sockaddr, int flags, struct unixcomm **newcommp)
 }
 
 static int
-unixcomm_listen(char *sockaddr, int flags, struct unixcomm **newcomm)
+unixcomm_listen(char *sockaddr, int flags, struct unixcomm **newcommp)
 {
 	struct sockaddr_un	address;
 	struct unixcomm		*comm;
 	int			ret;
 
-	*newcomm = NULL;
+	*newcommp = NULL;
 
 	ret = new_unixcomm_socket(flags, &comm);
 	if (ret != 0)
 		return ret;
 
 	fill_socket_address(sockaddr, &address);
-	errno = 0;
 	if (bind(comm->fd, (struct sockaddr *)&address,
 		 sizeof(struct sockaddr_un)) < 0) {
-		ret = daos_errno2der(errno);
 		D_ERROR("Failed to bind socket at '%.4096s', fd=%d, errno=%d\n",
 			sockaddr, comm->fd, errno);
 		unixcomm_close(comm);
-		return ret;
+		return -DER_MISC;
```

Comment:
Mind creating a ticket to have us look back into the errno issue later? I think it's just a matter of updating some unit tests and mocks.

Reply:
This was driving me mad earlier.  The mock code swaps out the libc functions so they appear to fail by returning -1, but the mock replacements to not set errno, so if the drpc code pre-sets errno to zero (which is a good idea), then call daos_errno2der(errno) errno is still zero, so these functions call all return DER_SUCCESS.  The only reason some of these tests pass is that errno is still set to something from some previous operation.  Bascially the previous commit was failing a bunch of unit tests so I had to back out some of the changes.

I suspect the correct fix would be to update the mock code to also set errno, and to have drpc initialise errno to zero and use daos_errno2der (as I did in a previous revision of this PR) but that's deeper down the rabbit hole than I intended.  I would be interested in input for people more familiar with the mock code to confirm all this however.

Reply:
Your analysis is correct. What we need for each of the libc mock functions is a way to tell it what to set the errno to (a new variable for each mock function, called something like bind_errno, socket_errno, etc), and to have each of the mocks set the errno to the corresponding value when called.

There are a few existing test cases where the errno is naively set by the test itself (rather than inside the mock), which seemed simpler at the time, but setting errno to 0 inside the production code breaks it.